frontend/src/shared/ui/layouts/MainLayout/MainLayout.tsx

```diff
@@ -1,31 +1,62 @@
 import { Suspense } from "react";
 import { Outlet, useLocation, useNavigate } from "react-router";
 
+import { useAuth } from "@/shared/auth";
+import { GearIcon, Pencil2Icon } from "@radix-ui/react-icons";
+
 import * as S from "./MainLayout.styles";
 
 export const MainLayout = () => {
   const navigate = useNavigate();
+  const { isAuthenticated, user } = useAuth();
   const { pathname } = useLocation();
 
   return (
     <S.Wrapper>
       <S.Header>
         <S.HeaderContainer>
           <S.HeaderLink href="#">conduit</S.HeaderLink>
-
-          <nav>
-            <S.HeaderNavList>
-              <S.HeaderNavItem $isActive={pathname === "/"} onClick={() => navigate("/")}>
-                Home
-              </S.HeaderNavItem>
-              <S.HeaderNavItem $isActive={pathname === "/signin"} onClick={() => navigate("/signin")}>
-                Sign in
-              </S.HeaderNavItem>
-              <S.HeaderNavItem $isActive={pathname === "/signup"} onClick={() => navigate("/signup")}>
-                Sign up
-              </S.HeaderNavItem>
-            </S.HeaderNavList>
-          </nav>
+          {isAuthenticated ? (
+            <nav>
+              <S.HeaderNavList>
+                <S.HeaderNavItem $isActive={pathname === "/"} onClick={() => navigate("/")}>
+                  Home
+                </S.HeaderNavItem>
+                <S.HeaderNavItem $isActive={pathname === "/editor"} onClick={() => navigate("/editor")}>
```

Comment:
since we have multiple routes, I would suggest to create constants for pathnames